api/types/gkesagenerator.go

```diff
@@ -0,0 +1,19 @@
+// Copyright 2019 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package types
+
+// GkeSaGeneratorArgs contains arguments to generate a GKE service account resource.
+type GkeSaGeneratorArgs struct {
+	// the name of the Kubernetes namespace for this object
+	K8sNamespace string `json:"k8sNamespace" yaml:"k8sNamespace"`
```

Comment:
Are you able to have nested fields here?  Because if so I was wondering about something like

```
...
kubernetesService:
  name:
  namespace:
...
```

Reply:
We can definitely have nested fields. 

What does the full resource that you'd like to generate look like?

Reply:
done